### PR TITLE
fix gtDispTree

### PR DESCRIPTION
**Pull request by sandreenko**
It is a legal situation when we pass ILNone (for example for local stores:
`gtDispChild(tree->gtOp.gtOp1, indentStack, IINone);`). In this code we
poped a value and didn't push anything back, it ended with the stack
underflow.

### CODE REVIEW

**Comment by sandreenko** [comment]
сс @dotnet/jit-contrib

**Comment by sandreenko** [comment]
I found it when adding my own debug `gtDispTree`, I have no idea why we do not hit it with JitDump.

Checked that JitDump works after this change.

**Comment by CarolEidt** [comment]
Can you give an example where this occurred?

File: src/jit/gentree.cpp [hunk 1]
```diff
@@ -10936,8 +10936,11 @@ void Compiler::gtDispTree(GenTreePtr   tree,
                 indentStack->Push(IIEmbedded);
                 lowerArc = IIEmbedded;
                 break;
+            case IINone:
+                indentStack->Push(IINone);
+                lowerArc = IINone;
```
Comment (rartemev) [inline src/jit/gentree.cpp]: I think you missed `break;`

Comment (sandreenko) [reply src/jit/gentree.cpp]: Thanks. Fixed,

**Comment by sandreenko** [comment]
> Can  you give an example where this occurred?

Yes, if you apply this patch: https://github.com/sandreenko/coreclr/commit/932c41e318ab1f59a3cb70fdf581257980a74ce2

and run corssgen with `complus_ngendump=IL_STUB_PInvoke` you will see an assert about the stack underflow, because it will try to print such tree:
```
before morph(
[000144] D-----------             *  STORE_LCL_VAR long   V11 FramesRoot
[000143] --C-G-------                *  CALL help long   HELPER.CORINFO_HELP_INIT_PINVOKE_FRAME
[000139] ------------ arg0           +--*  LCL_FLD_ADDR byref  V12 PInvokeFrame [+8]
[000140] ------------ arg1           \--*  PHYSREG   long   r10
```
and in the old version of the code we ends up in this `// Should never get here; just use IINone` , but the value was already poped from the stack.`

**Comment by CarolEidt** [comment]
If that is a "before morph" dump then there should not be a `GT_STORE_LCL_VAR` in the tree. Those are only added during `Lowering`. Before that, the front-end expects to see assignments.

**Comment by sandreenko** [comment]
> If that is a "before morph" dump then there should not be a GT_STORE_LCL_VAR in the tree. Those are only added during Lowering. Before that, the front-end expects to see assignments.

We call `morph` from `lowering` too.

**Comment by CarolEidt** [comment]
> We call morph from lowering too.

Yes, but the dump you show is dumping in pre-LIR form, where it's not expecting to see `GT_STORE_LCL_VAR` nodes. I suspect that the reason we didn't run into this before is that we wouldn't normally be dumping this kind of node when calling morph from `Lowering`. I think this is an OK fix, but I think we should always understand why something like this has only now become an issue, before we fix it.

**Comment by sandreenko** [comment]
> Yes, but the dump you show is dumping in pre-LIR form, where it's not expecting to see GT_STORE_LCL_VAR nodes. I suspect that the reason we didn't run into this before is that we wouldn't normally be dumping this kind of node when calling morph from Lowering. I think this is an OK fix, but I think we should always understand why something like this has only now become an issue, before we fix it.

Agree, thanks. 

When we dump LIR nodes we usually call `gtDispLIRNode` that calls `gtDispTree(node, &indentStack, nullptr, topOnly, isLIR);` with `topOnly == true`, it is why we do not print `lclStores` children.

**Pull request merged**